clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -6548,7 +6548,8 @@ ProcessDeclAttribute(Sema &S, Scope *scope, Decl *D, const ParsedAttr &AL,
                ? (unsigned)diag::err_keyword_not_supported_on_target
            : AL.isDeclspecAttribute()
                ? (unsigned)diag::warn_unhandled_ms_attribute_ignored
-               : (unsigned)diag::warn_unknown_attribute_ignored)
+           : AL.getScopeName() ? (unsigned)diag::ext_unknown_attribute_ignored
```

Comment:
This part isn't right, this changes our `clang::Baz` warning to an extension, when we only want to do it when we don't know what the namespace is.  We likely need to do some tablegen work to show the list of namespaces that we support, and issue the diagnostic there.